adding timeout to github client

loginakhil opened the pull request:
**Description**

The changes in this PR is to address issues found in https://github.com/hellofresh/jetstream/pull/379

boekkooi-fresh commented on github.go:
```diff
@@ -61,6 +62,7 @@ func NewGithubClient(
 	token string,
 	releaseFilter ReleaseFilter,
 	assetFilter AssetFilter,
+	connectionTimeout time.Duration,
```
It looks like this is never used

loginakhil replied:
Sorry, that was some IDE mess-up 🙈 
Fixed now 🙂

loginakhil commented:
> Shall we split this into 2 PR's? One is the move to `go mod` from `dep` and 1 being the timeout setting?

Moved `go mod` migration to another PR https://github.com/hellofresh/updater-go/pull/6

loginakhil commented:
> The timeout is currently only applied when a token is provided which is often not the case for public repo's

Good catch, i have changed the implementation now. Please take a new look 🙏

boekkooi-fresh commented on github.go:
```diff
@@ -61,6 +62,7 @@ func NewGithubClient(
 	assetFilter AssetFilter,
+	connectionTimeout time.Duration,
 ) *GithubLocator {
 	var httpClient *http.Client
```
Since we don't allow specifying the http client set the timeout directly?
```suggestion
	httpClient := &http.Client{Timeout: connectionTimeout}
```

loginakhil replied:
If we do this, we still have to set timeout like this

```go
httpClient.Timeout = connectionTimeout
```

for `oauth2 client here`

```go
httpClient = oauth2.NewClient(context.Background(), src)
```

wdyt ?

boekkooi-fresh replied:
Good point. So probably something like this is the best to also not effect the default http client.
```
	var httpClient *http.Client
	if token != "" {
		src := oauth2.StaticTokenSource(
			&oauth2.Token{AccessToken: token},
		)
		httpClient = oauth2.NewClient(context.Background(), src)
		httpClient.Timeout = connectionTimeout
	} else {
		httpClient = &http.Client{Timeout: connectionTimeout}
	}
```

loginakhil replied:
Thanks, I have updated the behaviour now.

boekkooi-fresh commented:
:+1:

ankitpokhrel commented:
+1

olvlvl commented:
+1
